Upgrading xarray to > 2023.12.0 fails CI tests

Dependabot PR #530 bumps xarray to 2024.3.0. This apparently causes the package to fail its tests in CI (logs from the CI run are attached).

```
=========================== short test summary info ============================
FAILED src/impactlab_tools/gcp/reindex.py::impactlab_tools.gcp.reindex.hierid_to_shapenum
FAILED src/impactlab_tools/gcp/reindex.py::impactlab_tools.gcp.reindex.shapenum_to_hierid
=================== 2 failed, 31 passed, 2 warnings in 6.23s ===================
```

Looks like it might just be a change in repr/str formatting of DataArrays or Dataset from what we have in docstrs.


[logs_23349733380.zip](https://github.com/user-attachments/files/15506683/logs_23349733380.zip)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -62,6 +62,8 @@ exclude = [
     ".eggs",
     "docs",
 ]
+
+[tool.ruff.lint]
 # E402: module level import not at top of file
 ignore = [
     "E402",

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,6 @@
 # This is only used by Github and Github Actions.
 numpy==1.26.3
-xarray==2023.12.0
+xarray==2024.3.0
 pandas==2.2.1
 scipy==1.13.0
 h5netcdf==1.3.0

--- a/src/impactlab_tools/gcp/reindex.py
+++ b/src/impactlab_tools/gcp/reindex.py
@@ -70,12 +70,12 @@ def shapenum_to_hierid(data, dim='SHAPENUM', new_dim='hierid', inplace=False):
         ...
         >>> reshaped = shapenum_to_hierid(ds)
         >>> reshaped # doctest: +ELLIPSIS
-        <xarray.Dataset>
+        <xarray.Dataset> Size: 4MB
         Dimensions:  (hierid: 24378)
         Coordinates:
-          * hierid   (hierid) ...
+          * hierid   (hierid) <U35 3MB 'CAN.1.2.28' 'CAN.1.17.403' ... 'BWA.4.13'
         Data variables:
-            var1     (hierid) float64 0.417 0.7203 0.0001144 ...
+            var1     (hierid) float64 195kB 0.417 0.7203 0.0001144 ... 0.1359 0.9259
 
         >>> (reshaped.var1.values == ds.var1.values).all()
         True
@@ -151,12 +151,12 @@ def hierid_to_shapenum(data, dim='hierid', new_dim='SHAPENUM', inplace=False):
         ...
         >>> reshaped = hierid_to_shapenum(ds)
         >>> reshaped
-        <xarray.Dataset>
+        <xarray.Dataset> Size: 160B
         Dimensions:   (SHAPENUM: 10)
         Coordinates:
-          * SHAPENUM  (SHAPENUM) float64 1.369e+03 5.747e+03 ...
+          * SHAPENUM  (SHAPENUM) float64 80B 1.369e+03 5.747e+03 ... 5.812e+03 5.832e+03
         Data variables:
-            var2      (SHAPENUM) float64 0.417 0.7203 0.0001144 ...
+            var2      (SHAPENUM) float64 80B 0.417 0.7203 0.0001144 ... 0.3968 0.5388
 
         >>> (reshaped.var2.values == ds.var2.values).all()
         True

--- a/src/impactlab_tools/utils/paralog.py
+++ b/src/impactlab_tools/utils/paralog.py
@@ -121,7 +121,7 @@ class StatusManager:
                 fp.write(status + '\n')
         except Exception:
             print("CAUGHT A WILD EXCEPTION BUT IGNORING IT WITHOUT LOGGING IT!")
-            print("Warning: Could write status update %s" % status)
+            print(f"Warning: Could write status update {status}")
 
     def release(self, dirpath, status):
         """Release the claim on this directory."""
@@ -136,7 +136,7 @@ class StatusManager:
                 fp.write(f"{time.asctime()} {self.jobtitle}: {status}\n")
         except Exception:
             print("CAUGHT A WILD EXCEPTION BUT IGNORING IT WITHOUT LOGGING IT!")
-            print("Warning: Could write release status %s" % status)
+            print(f"Warning: Could write release status {status}")
 
     def is_claimed(self, dirname):
         """Check if a directory has claims from any of our jobs."""
@@ -172,7 +172,7 @@ class StatusManager:
     @staticmethod
     def claiming_filepath(dirpath, jobname):
         """Return the path to the status file used for claiming a directory."""
-        return os.path.join(dirpath, "status-%s.txt" % jobname)
+        return os.path.join(dirpath, f"status-{jobname}.txt")
 
     @staticmethod
     def kill_active(dirpath, jobname):

--- a/whatsnew.rst
+++ b/whatsnew.rst
@@ -6,6 +6,10 @@ These are new features and improvements of note in each release.
 Unreleased
 ----------
 
+ - Minor code cleanup, style update.
+
+ - Update ruff lint section format in pyproject.toml
+
 v0.5.0 (August 23, 2023)
 ------------------------
 

--- a/whatsnew.rst
+++ b/whatsnew.rst
@@ -10,6 +10,8 @@ Unreleased
 
  - Update ruff lint section format in pyproject.toml
 
+ - Minor update to docstr examples to match xarray > 2023.12.0.
+
 v0.5.0 (August 23, 2023)
 ------------------------
 

